homeassistant/components/advantage_air/climate.py

```diff
@@ -82,39 +83,116 @@ async def async_setup_entry(
 
     @property
-    def target_temperature(self):
+    def target_temperature(self) -> float | None:
         """Return the current target temperature."""
         return self._ac["setTemp"]
 
     @property
-    def hvac_mode(self):
+    def hvac_mode(self) -> HVACMode:
         """Return the current HVAC modes."""
         if self._ac["state"] == ADVANTAGE_AIR_STATE_ON:
-            return ADVANTAGE_AIR_HVAC_MODES.get(self._ac["mode"])
+            return ADVANTAGE_AIR_HVAC_MODES[self._ac["mode"]]
         return HVACMode.OFF
 
     @property
-    def fan_mode(self):
+    def hvac_modes(self) -> list[HVACMode]:
+        """Return the available HVAC modes."""
+        # MyTemp only supports cooling and heating
+        if self._ac.get(ADVANTAGE_AIR_MYTEMP_ENABLED):
+            return [HVACMode.OFF, HVACMode.COOL, HVACMode.HEAT]
+        # MyAuto adds support for Auto
+        if self._ac.get(ADVANTAGE_AIR_MYAUTO_ENABLED):
+            return [
+                HVACMode.OFF,
+                HVACMode.COOL,
+                HVACMode.HEAT,
+                HVACMode.FAN_ONLY,
+                HVACMode.DRY,
+                HVACMode.HEAT_COOL,
+            ]
+        # MyZone does not support auto
+        return [
+            HVACMode.OFF,
+            HVACMode.COOL,
+            HVACMode.HEAT,
+            HVACMode.FAN_ONLY,
+            HVACMode.DRY,
+        ]
+
+    @property
+    def fan_mode(self) -> str | None:
         """Return the current fan modes."""
         return ADVANTAGE_AIR_FAN_MODES.get(self._ac["fan"])
 
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the list of available fan modes."""
+        # Auto is only available when AutoFan is enabled
+        if self._ac.get(ADVANTAGE_AIR_AUTOFAN):
+            return self._attr_fan_modes + [FAN_AUTO]
+        return self._attr_fan_modes
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode."""
+        if self._ac.get(ADVANTAGE_AIR_MYAUTO_ENABLED):
+            return ADVANTAGE_AIR_MYAUTO
+        if self._ac.get(ADVANTAGE_AIR_MYTEMP_ENABLED):
+            return ADVANTAGE_AIR_MYTEMP
+        return ADVANTAGE_AIR_MYZONE
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        # MyTemp does not support setting a temperature of any kind
+        if self.preset_mode == ADVANTAGE_AIR_MYTEMP:
+            return self._attr_supported_features
+
+        # MyAuto in Heat/Cool supports setting a temperature range
+        if self.hvac_mode == HVACMode.HEAT_COOL:
+            return (
+                self._attr_supported_features
+                | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            )
+
+        # MyAuto and MyZone in any other mode supports a setting a target temperature
+        return self._attr_supported_features | ClimateEntityFeature.TARGET_TEMPERATURE
```

Comment:
@bdraco is this also going to be unacceptable? I cant have the intergration accepting both TARGET_TEMPERATURE_RANGE and TARGET_TEMPERATURE at the same time as it will be unusable in the UI.

@balloob is what I am doing just not possible in Home Assistant?

Reply:
Supported features isn't allowed to change at runtime either